beacon-chain/state/state-native/setters_withdrawal.go

```diff
@@ -84,3 +98,50 @@ func (b *BeaconState) SetNextPartialWithdrawalValidatorIndex(i types.ValidatorIn
 	b.nextPartialWithdrawalValidatorIndex = i
 	return nil
 }
+
+// WithdrawBalance withdraws the balance from the validator and creates a
+// withdrawal receipt for the EL to process
+func (b *BeaconState) WithdrawBalance(index types.ValidatorIndex, amount uint64) error {
```

Comment:
Is a lock needed for this mutator method?

Reply:
Nice catch!

Reply:
No I didn't think so as all access are self locked, can check in an hour

Reply:
Yeah, i noticed that as well. It gives me some concern about race conditions where values could be updated during this mutator call.